### PR TITLE
[semver:minor] Implement Deletion Threshold Checks for Users and Groups

### PR DESCRIPTION
**Pull request by khush-healx**
https://healxltd.atlassian.net/jira/software/projects/ENT/boards/105?selectedIssue=ENT-421

This PR introduces new functionality to the `ssosync` tool to prevent accidental large-scale deletions of users and groups. We've added two functions, `checkUserDeletionThreshold` and `checkGroupDeletionThreshold`, which enforce a maximum threshold for deletions per operation. The threshold is set at 4 deletions. If a deletion operation exceeds this number, the operation is halted, and a warning is logged. This update aims to enhance the robustness of our user and group synchronization process by adding a safeguard against potential sync issues or misconfigurations.

**Changes:**
1. Added `checkUserDeletionThreshold` function to limit user deletions.
2. Added `checkGroupDeletionThreshold` function to limit group deletions.
3. Integrated these functions into the existing deletion workflows to check the threshold before proceeding with deletions.

### CODE REVIEW

**Review by gareth-rogers-healx — COMMENTED** [review]
I think as this is written the code is fine, I wonder if we should lower the thresholds, lets chat in Slack with @alexhermida 

My comment is to remove the comments in the function, they don't add anything that isn't obvious from the code.

I think either delete the warning message as it's logged in the calling code. The message as written doesn't quite make sense as it's a check and no deletion occurs within the function itself.

If we keep it I think it should be more like: "Given %d groups that would be deleted, this exceeds the threshold of %d groups."

If the warning is deleted you can just return the condition I think `return len(groups) > deletionThreshold`.

File: internal/sync.go [hunk 1]
```diff
@@ -908,3 +916,23 @@ func (s *syncGSuite) includeGroup(name string) bool {
 
 	return false
 }
+
+// checkUserDeletionThreshold checks if the number of users to be deleted exceeds the threshold.
+func checkUserDeletionThreshold(users []*aws.User) bool {
+    const deletionThreshold = 4 // Define the maximum number of users that can be deleted at once.
```
Comment (alexhermida) [inline internal/sync.go]: maybe setting this as a `var` at the top of the file?

File: internal/sync.go [hunk 2]
```diff
@@ -402,6 +402,10 @@ func (s *syncGSuite) SyncGroupsUsers(query string) error {
 	log.Info("syncing changes")
 	// delete aws users (deleted in google)
 	log.Debug("deleting aws users deleted in google")
+	if !checkUserDeletionThreshold(delAWSUsers) {
```
Comment (alexhermida) [inline internal/sync.go:405]: Maybe this and the other function are redundant as just having just the if here is enough

```go
if len(delAWSUsers) > deletionThreshold { ... }
```

**Review by alexhermida — COMMENTED** [review]
I made a couple of comments, in similar lines. I think less code the better 😄  in this case looks like with just an `if` it's enough. But happy to chat!

**Review by gareth-rogers-healx — APPROVED** [review]
🦁

**Pull request merged**